Content.Server/Station/Systems/StationSystem.cs

```diff
@@ -0,0 +1,406 @@
+    {
+        if (!Resolve(station, ref stationData, ref metaData))
+            throw new ArgumentException("Tried to use a non-station entity as a station!", nameof(station));
+
+        var oldName = metaData.EntityName;
+        metaData.EntityName = name;
+
+        if (loud)
+        {
+            _chatManager.DispatchStationAnnouncement($"The station {oldName} has been renamed to {name}.");
+        }
+
+        RaiseLocalEvent(station, new StationRenamedEvent(oldName, name));
+    }
+
+    /// <summary>
+    /// Deletes the given station.
+    /// </summary>
+    /// <param name="station">Station to delete.</param>
+    /// <param name="stationData">Resolve pattern, station data component of station.</param>
+    /// <exception cref="ArgumentException">Thrown when the given station is not a station.</exception>
+    public void DeleteStation(EntityUid station, StationDataComponent? stationData = null)
+    {
+        if (!Resolve(station, ref stationData))
+            throw new ArgumentException("Tried to use a non-station entity as a station!", nameof(station));
+
+        _stations.Remove(station);
+        Del(station);
+    }
+
+    /// <summary>
+    /// Gets the station that "owns" the given entity (essentially, the station the grid it's on is attached to)
+    /// </summary>
+    /// <param name="entity">Entity to find the owner of.</param>
+    /// <param name="xform">Resolve pattern, transform of the entity.</param>
+    /// <returns>The owning station, if any.</returns>
+    /// <remarks>
+    /// This does not remember what station an entity started on, it simply checks where it is currently located.
+    /// </remarks>
+    public EntityUid? GetOwningStation(EntityUid entity, TransformComponent? xform = null)
+    {
+        if (!Resolve(entity, ref xform))
+            throw new ArgumentException("Tried to use an abstract entity!", nameof(entity));
+
+        if (TryComp<IMapGridComponent>(entity, out _))
+        {
+            // We are the station, just check ourselves.
+            return CompOrNull<StationMemberComponent>(entity)?.Station;
+        }
+
+        if (xform.GridID == GridId.Invalid)
+        {
+            Logger.Debug("A");
+            return null;
+        }
+
+        var grid = _mapManager.GetGridEuid(xform.GridID);
+
+        return CompOrNull<StationMemberComponent>(grid)?.Station;
+    }
+}
+
+/// <summary>
+/// Event fired when a station is first set up.
+/// This is the ideal point to add components to it.
+/// </summary>
+[PublicAPI]
+public sealed class StationInitializedEvent : EntityEventArgs
+{
+    /// <summary>
+    /// Station this event is for.
+    /// </summary>
+    public EntityUid Station;
+
+    public StationInitializedEvent(EntityUid station)
+    {
+        Station = station;
+    }
+}
+
+/// <summary>
+/// Event fired when a grid becomes a member of a station.
+/// </summary>
```

Comment:
docs should specify if its broadcast or directed and on which ent, i assume this is directed on the station?